### DISCUSSION TITLE
Numpy not available

### DISCUSSION THREAD

**Discussion by Nelsoncrib** [openai/whisper#1032] [Q&A] [answered]
I tried running the model, but I keep getting `RuntimeError: Numpy is not available ` 

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [40], in <cell line: 10>()
      7 audio = whisper.pad_or_trim(audio)
      9 # make log-Mel spectrogram and move to the same device as the model
**---> 10 mel = whisper.log_mel_spectrogram(audio).to(model.device)**
     12 # detect the spoken language
     13 _, probs = model.detect_language(mel)

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\whisper\audio.py:112, in log_mel_spectrogram(audio, n_mels)
    110     if isinstance(audio, str):
    111         audio = load_audio(audio)
**--> 112     audio = torch.from_numpy(audio)**
    114 window = torch.hann_window(N_FFT).to(audio.device)
    115 stft = torch.stft(audio, N_FFT, HOP_LENGTH, window=window, return_complex=True)

**RuntimeError: Numpy is not available**

**Comment by jongwook** [comment, marked as answer]
Could you try:

```
pip install numpy
```

and running it again?

**Reply by Nelsoncrib** [reply]
Thank you for your response. I already tried pip install numpy. I have the updated Numpy but it wouldn't just work with the model. I do not know why.

**Reply by jongwook** [reply]
Strange. Which version of numpy are you using? `import numpy; print(numpy.__version__)`

**Reply by Nelsoncrib** [reply]
1.22.4

**Reply by jongwook** [reply]
I think I found the cause - [PyTorch on Windows does not support Python 3.10](https://pytorch.org/get-started/locally/#windows-python), so you'll need to downgrade to Python 3.9 for the time being:

- https://github.com/coqui-ai/TTS/issues/2076#issuecomment-1279790572
- https://discuss.pytorch.org/t/numpy-is-not-available-error/146192/2

**Reply by Nelsoncrib** [reply]
Thank you for this. One last thing, can you share a link where I can download the 3.9 version? The website is not giving me the room to download a lower version. Thanks.

**Reply by jongwook** [reply]
I think it'll be more convenient to use the anaconda installer: https://www.anaconda.com/products/distribution#Downloads it comes prepackaged with most of the Python packages you need, along with the `conda` command that you can use to install PyTorch as in https://pytorch.org/get-started/locally/

**Reply by Nelsoncrib** [reply]
Thank you for this. I'm grateful.

**Reply by virtualcrane** [reply]
I'm on Mac. I've gotten the same error. Does this fix apply to me?
